### DISCUSSION TITLE
Capture all internal cFS CCSDS traffic?

### DISCUSSION THREAD

**Discussion by joshua99911** [nasa/cFS#733] [Q&A] [answered]
Hi all, **I'm working on a project that requires creating a dataset of telecommand (CI) form a ground station, telemetry (TO) from a satellite and internal traffic capture.**

**I have the first two achieved by capturing the data over their specified ports but am struggling with any internal traffic** of the cFS that is not expressed in CI and TO traffic. **As far as I am aware the internal traffic is configured as CCSDS space packets and I am currently using the Datastore (DS) application but want to know if this is the correct approach** as my aim is to have a dataset of CCSDS traffic that I can classify as telecommand, telemetry and internal traffic for later use.

Any help would be greatly appreciated in steering me in the right direction.

**Comment by irowebbn** [comment, marked as answer]
Yes, you are correct that cFE Message packets are implemented as CCSDS Space Packets.

The approach you mention is what I have seen as well. Take a look at [`fsw/inc/ds_extern_typedefs.h`](https://github.com/nasa/DS/blob/b540e3be34d2a34d4ef85ff3442a49e3ccbeb6b3/fsw/inc/ds_extern_typedefs.h) in the DS app. There are two tables that you load into the app in order to store messages- the Destination File table and the Filter table. The destination file table simply information about one or more files you want to use to organize and store your data. The filter table takes entries of message IDs (telecommand or telemetry) with some parameters about how often to log them. You can enter the same message ID into the table multiple times if you want to store it in multiple different files. 

The message ID already encodes whether a packet is a command or telemetry, but you could also use multiple files for your classification.
